rust/ql/lib/codeql/rust/controlflow/internal/Completion.qll

```diff
@@ -117,13 +117,61 @@ class BooleanCompletion extends ConditionalCompletion, TBooleanCompletion {
   override string toString() { result = "boolean(" + value + ")" }
 }
 
+/** Holds if node `pat` has the constant match value `value`. */
```

Comment:
I don't think this is correct use of "irrefutable". Irrefutable is property of a pattern, but this predicate is a property of a pattern as an AST node and considers the surrounding AST context as well.

For instance, in
```rust
match b { Some(a) => 1, None => 0 }
```
this predicate holds for `None` (since the last pattern in a `match` arm is guaranteed to succeed due to exhaustiveness) but `None` is not an irrefutable pattern.

I think we should rename this predicate (not sure to what though). We could also extract a proper `isIrrefutablePattern` predicate from this predicate (but it will probably be a conservative approximation as refutability for custom types depends on the type definition. E.g. the pattern `Foo(a)` is irrefutable if `Foo` the only case in the `enum` and refutable otherwise).

Reply:
Yeah, you're right. Perhaps `isExhaustiveMatch` or similar. An irrefutable pattern is exhaustive on its own, and a last pattern in `match` should be exhaustive.

Reply:
That sounds like a great name to me. I'll do a PR :)

Reply:
Done, PR is here https://github.com/github/codeql/pull/17751